VultisigApp/VultisigApp/Model/FunctionCall/FunctionCallAddThorLP.swift

```diff
@@ -0,0 +1,524 @@
+                    return
+                }
+                
+                if inbound.halted || inbound.global_trading_paused || inbound.chain_trading_paused || inbound.chain_lp_actions_paused {
+                    return
+                }
+                
+                let destinationAddress: String
+                if tx.coin.shouldApprove {
+                    // ERC20 token (e.g., USDC) → approval to router
+                    destinationAddress = inbound.router ?? inbound.address
+
+                } else {
+                    // Native token (e.g., ETH) → direct to inbound address
+                    destinationAddress = inbound.address
+
+                }
+                
+                tx.toAddress = destinationAddress
+                
+                // ERC20 approval only for non-RUNE ERC20 tokens
+                isApprovalRequired = tx.coin.shouldApprove
+                if isApprovalRequired {
+                    if !tx.toAddress.isEmpty {
+                        let payload = ERC20ApprovePayload(amount: tx.amountInRaw, spender: tx.toAddress)
+                        self.approvePayload = payload
+
+                    } else {
+                        Task {
+                            for attempt in 0..<5 {
+                                try? await Task.sleep(nanoseconds: 300_000_000)
+                                if !tx.toAddress.isEmpty {
+                                    let payload = ERC20ApprovePayload(amount: tx.amountInRaw, spender: tx.toAddress)
+                                    await MainActor.run {
+                                        self.approvePayload = payload
+                                    }
+                                    break
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    // MARK: - Pool loading
+    
+    private func cleanPoolName(_ asset: String) -> String {
+        if let dashIndex = asset.firstIndex(of: "-") {
+            let suffix = asset[asset.index(after: dashIndex)...]
+            if suffix.uppercased().starts(with: "0X") {
+                return String(asset[..<dashIndex])
+            }
+        }
+        return asset
+    }
+    
+    @MainActor
+    func loadPools() async {
+        isLoadingPools = true
+        loadError = nil
+        
+        let timeoutTask = Task {
+            try? await Task.sleep(nanoseconds: 15_000_000_000) // 15s
+            if isLoadingPools {
+                await MainActor.run {
+                    self.availablePools = []
+                    self.isLoadingPools = false
+                    self.loadError = "Timeout loading pools."
+                }
+            }
+        }
+        
+        do {
+            let allPools = try await Task.detached {
+                try await ThorchainService.shared.fetchLPPools()
```

Comment:
only `ThorchainService.shared.fetchLPPools()` could fail because network , why not just add retry for this metho , instead of the whole thing?